pkg/apis/core/v1beta1/types_shoot.go

```diff
@@ -141,6 +141,9 @@ type ShootSpec struct {
 	// AccessRestrictions describe a list of access restrictions for this shoot cluster.
 	// +optional
 	AccessRestrictions []AccessRestrictionWithOptions `json:"accessRestrictions,omitempty" protobuf:"bytes,24,rep,name=accessRestrictions"`
+	// ETCD contains configuration for etcds of the shoot cluster.
+	// +optional
+	ETCD *ETCD `json:"etcd,omitempty" protobuf:"bytes,25,opt,name=etcd"`
```

Comment:
Aren't `.spec.controlPlane` or `.spec.kubernetes` better locations for the `etcd` field instead of having it at the root level in the Shoot spec?

Reply:
Good question! I wasn't too sure about this:

1. `.spec.controlPlane` seems more like a general settings section, otherwise you could argue that `kubernetes` belongs to `.spec.controlPlane`, too.
2. `.spec.kubernetes` is inconsistent to how the `Garden` resource is structured. There we have `.spec.virtualCluster.etcd` not `.spec.virtualCluster.kubernetes.etcd`

Based on these thoughts I think it's fine to have `.spec.etcd`, but if I had to choose between 1. and 2., I'd rather go with option 1.

Any opinions?

Reply:
It's easier to adapt the `Garden` API than to adapt the `Shoot` API. I'm also in favor with putting it under `.spec.controlPlane` or `.spec.kubernetes` instead of introducing it on top-level.

Reply:
Ok, makes sense. Thank you!